src/vm/array.cpp

```diff
@@ -250,17 +250,6 @@ MethodTable* Module::CreateArrayMethodTable(TypeHandle elemTypeHnd, CorElementTy
     // Shared EEClass if there is one
     MethodTable * pCanonMT = NULL;
 
-    // Strictly speaking no method table should be needed for
-    // arrays of the faked up TypeDescs for variable types that are
-    // used when verfifying generic code.
-    // However verification is tied in with some codegen in the JITs, so give these
-    // the shared MT just in case.
-    // This checks match precisely one in ParamTypeDesc::OwnsTemplateMethodTable
-    if (CorTypeInfo::IsGenericVariable(elemType)) {
-        // This is loading the canonical version of the array so we can override
-        OVERRIDE_TYPE_LOAD_LEVEL_LIMIT(CLASS_LOADED);
-        return(ClassLoader::LoadArrayTypeThrowing(TypeHandle(g_pObjectClass), arrayKind, Rank).GetMethodTable());
```

Comment:
It would be worth looking at how many extra MethodTables get created for a typical app (e.g. the default ASP.NET webapp). Just so that we know how much we are paying for this simplification.

Reply:
Yes. I will try to get some estimates on how often `T[]` types could be created.

Reply:
I checked how many of these are created by FX tests (since that I can do easily)

==== System.Linq.Expression tests:
Before starting the actual testing creates
 124 array types, of which 9  are open generic instantiations

I guess that is just jitting xunit stuff. Hitting the first instantiations of `Enumerable<T>.Select`, `ToArray` and like that ...

At the end of all Expression tests 
 Total: 676, Generic: 73

=====  System.Linq
 Total: 363, Generic: 35

=====  System.Linq.Queryable
 Total: 292, Generic: 20

=====  Microsoft.CSharp
 Total: 731, Generic: 18        <-  running actual tests adds just 9, I expected more from this one, it does lots of generics. I guess not a lot of arrays

====  System.Runtime.Metadata
 Total: 326, Generic: 20

==== System.Websocket.Client
Total: 225, Generic: 16

Overall it is a bit more than I expected. 
I expected nearly none at all (except for `S.L.Expr`), since there are no instances.

Most of this comes from jit validation - like `CanAccessClass` and `getCallInfo`. There are places where validation must happen against open instantiations for correctness.

However this does not seem huge numbers either. It looks like depending on genericness of the code, open generic method tables can be about 5%-10% of total array MTs.

If we stop allocating TypeDecs for arrays, we may get the space back.

I wonder how big these things are.

Reply:
Adding a new `T[]` takes 900+ bytes or so from the loader heap in bits and pieces. Hmm!

I suppose this is still ok for the gain of consistency, since the overall number is small, relative to other types.

Reply:
I am not sure how much to trust corefx tests to see the impact on the real-world apps. I would like to know how many of these we will create during startup of typical ASP.NET app and WPF app.

Reply:
Do we have a "typical" ASP.NET and WPF apps somewhere that we use for this kind of measuring?

Reply:
I typically use `dotnet new webapp` or `dotnet new wpf`

Make sure that the shared framework is R2R. I typically crossgen the whole app - the app itself is small so it does not make a huge difference.